network-mux/demo/cardano-ping.hs

```diff
@@ -0,0 +1,342 @@
+{-# LANGUAGE BangPatterns      #-}
+{-# LANGUAGE DataKinds         #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RecordWildCards   #-}
+{-# LANGUAGE RankNTypes        #-}
+
+module Main (main) where
+
+import qualified Codec.CBOR.Encoding as CBOR
+import qualified Codec.CBOR.Decoding as CBOR
+import qualified Codec.CBOR.Read     as CBOR
+import qualified Codec.CBOR.Write    as CBOR
+import           Control.Exception
+import           Control.Monad (replicateM, when, unless)
+import           Control.Monad.Class.MonadAsync
+import           Control.Monad.Class.MonadTime
+import           Control.Monad.Class.MonadTimer hiding (timeout)
+import           Control.Tracer (nullTracer)
+import           Data.Aeson
+import           Data.ByteString.Lazy (ByteString)
+import qualified Data.ByteString.Lazy as BL
+import qualified Data.ByteString.Lazy.Char8 as C8 (putStrLn)
+import           Data.Maybe (fromMaybe)
+import           Data.Word
+import           Data.TDigest
+import           Data.Text (unpack)
+import           Network.Socket ( AddrInfo)
+import qualified Network.Socket as Socket
+import           Text.Printf
+import           System.Environment (getArgs)
+import           System.Console.GetOpt
+import           System.IO (hFlush, hPutStr, stderr, stdout)
+
+import           Network.Mux.Bearer.Socket
+import           Network.Mux.Types
+import           Network.Mux.Timeout
+
+mainnetMagic :: Word32
+mainnetMagic = 764824073
+
+handshakeNum ::  MiniProtocolNum
+handshakeNum = MiniProtocolNum 0
+
+keepaliveNum :: MiniProtocolNum
+keepaliveNum = MiniProtocolNum 8
+
+data Flag = HostF String | PortF String | MagicF String | QuietF | JsonF deriving Show
+
+options :: [OptDescr Flag]
+options = [
+    Option "h" ["h"]     (ReqArg HostF "host")  "hostname/ip, e.g relay.iohk.example",
```

Comment:
Just neat picking: I'd use`['h']` rather than `"h"`.   I would also make the long option `--host` rather than `--h`, there's already `-h`.

Reply:
Will fix the long option.